test-cloud/test-cloud-performance/src/test/java/org/kie/cloud/integrationtests/s2i/KieServerS2iJbpmEJBTimerPerfIntegrationTest.java

```diff
@@ -0,0 +1,368 @@
+                scenarios.add(new Object[] { "KIE Server HTTPS S2I", kieServerHttpsS2ISettings });
+            }
+        } catch (UnsupportedOperationException ex) {
+            logger.info("KIE Server HTTPS S2I is skipped.", ex);
+        }
+
+        return scenarios;
+    }
+    
+    protected KieServicesClient kieServicesClient;
+    protected ProcessServicesClient processServicesClient;
+    protected UserTaskServicesClient taskServicesClient;
+    
+    protected QueryServicesClient queryServicesClient;
+
+    protected String repositoryName;
+    
+    protected List<String> pods = new ArrayList<String>();
+    
+    protected String startingTime, processTime;
+    protected Map<String, Integer> completedHostNameDistribution;
+
+    @Override
+    protected GenericScenario createDeploymentScenario(DeploymentScenarioBuilderFactory deploymentScenarioFactory) {       
+        logger.info("git provider:" + Git.getProvider());
+        repositoryName = Git.getProvider().createGitRepositoryWithPrefix("KieServerS2iJbpmRepository", ClassLoader.class.getResource(PROJECT_SOURCE_FOLDER).getFile());
+
+        DeploymentSettings kieServerS2Isettings = kieServerS2ISettingsBuilder
+                .withContainerDeployment(KIE_CONTAINER_DEPLOYMENT)
+                .withSourceLocation(Git.getProvider().getRepositoryUrl(repositoryName), REPO_BRANCH, DEFINITION_PROJECT_NAME)
+                .withTimerServiceDataStoreRefreshInterval(Duration.ofSeconds(REFRESH_INTERVAL))
+                .withKieServerMemoryLimit(HEAP)
+                .build();
+
+        return deploymentScenarioFactory.getGenericScenarioBuilder()
+                .withKieServer(kieServerS2Isettings)
+                .build();
+    }
+
+    @Before
+    public void setUp() {
+        
+        if (SCALE_COUNT > 1) {
+            logger.info("starting to scale");
+            scaleKieServerTo(SCALE_COUNT);
+            logger.info("scaled to {} pods", SCALE_COUNT);
+        } else if (SCALE_COUNT == 1) {
+            pods.add(deploymentScenario.getKieServerDeployments().get(0).getInstances().get(0).getName());
+        } else throw new RuntimeException("wrong scale parameter, should be equal or greater than 1");
+        
+        
+        deploymentScenario.getKieServerDeployments().get(0).setRouterTimeout(Duration.ofMinutes(ROUTER_TIMEOUT));
+        deploymentScenario.getKieServerDeployments().get(0).setRouterBalance(ROUTER_BALANCE);
+        
+        Map<String, String> requests = new HashMap<String, String>();
+        requests.put(CPU, System.getProperty("requests.cpu","1000m"));
+        requests.put(MEMORY, System.getProperty("requests.memory","1Gi"));
+        Map<String, String> limits = new HashMap<String, String>();
+        limits.put(CPU, System.getProperty("limits.cpu","4000m"));
+        limits.put(MEMORY, System.getProperty("limits.memory","4Gi"));
+        deploymentScenario.getKieServerDeployments().get(0).setResources(requests, limits);
```

Comment:
to be sure change is done and pod is ready, it would be also good to add there waiter for Kie Server to be ready. something like: deploymentScenario.getKieServerDeployments().get(0).waitForScale()

Reply:
ok, I wasn't aware of changes of the DC implies that pods are also restarted, so we need to wait for this to be ready
Done, and added also a comment to reflect this. Thx!